histogram/histogram_test.go

```diff
@@ -0,0 +1,109 @@
+package histogram
+
+import (
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+func Example() {
+	// Create a new descriptor.
+	desc := prometheus.NewDesc(
+		"foo_bar_metric",
+		"The number of foos that have been bared",
+		nil,
+		nil,
+	)
+
+	// Create a Histogram.
+	c := Config{
+		BucketLimits: []float64{0, 1, 5},
+	}
+	h, _ := New(c)
+
+	// Add entries to the Histogram as part of the exporter,
+	// e.g: latency of network connections.
+	h.Add(0.2)
+
+	// Convert the Histogram into a metric.
+	metric := prometheus.MustNewConstHistogram(
+		desc, h.Count, h.Sum, h.Buckets,
+	)
+
+	fmt.Println(metric)
+}
+
+func Test_Add(t *testing.T) {
+	testCases := []struct {
+		name            string
+		config          Config
+		addInput        []float64
+		expectedBuckets map[float64]uint64
+	}{
+		{
+			name: "case 0: Add 1 input to histogram with one bucket",
+			config: Config{
+				BucketLimits: []float64{5},
+			},
+			addInput: []float64{1},
+			expectedBuckets: map[float64]uint64{
+				5: 1,
+			},
+		},
+
+		{
+			name: "case 1: Add 1 input between buckets to a histogram with 2 buckets",
+			config: Config{
+				BucketLimits: []float64{5, 10},
+			},
+			addInput: []float64{6},
+			expectedBuckets: map[float64]uint64{
+				5:  0,
+				10: 1,
+			},
+		},
+
+		{
+			name: "case 2: Add 2 inputs for a histogram with 2 buckets",
+			config: Config{
+				BucketLimits: []float64{5, 10},
+			},
+			addInput: []float64{1, 7},
+			expectedBuckets: map[float64]uint64{
+				5:  1,
+				10: 2,
+			},
+		},
+
+		{
+			name: "case 3: Add 1 input higher than the bucket",
+			config: Config{
+				BucketLimits: []float64{10},
```

Comment:
So what happens when I configure `0` or `-500`?

Reply:
It's weird, but it should work. I'll add a test.

Reply:
Done